crates/orderbook-client/src/lib.rs

```diff
@@ -25,14 +24,15 @@ pub fn subscribe(url: String) -> impl Stream<Item = Result<String, Error>> + Unp
 /// It subscribes and yields all messages.
 pub fn subscribe_with_authentication(
     url: String,
-    credentials: Credentials,
+    authenticate: &dyn Fn(Message) -> Signature,
```

Comment:
❓ Asking because I'm never quite sure: could this be `impl Fn(Message) -> Signature` instead? I might give it a go myself to check.

Reply:
The compiler was recommending this as it was complaining about the function not being Sized. 
Give it a try and let me know. I'll wait with merging this :)

Reply:
It just worked with `impl` for me, so I force-pushed the amended commit 🤷

Reply:
mhmmm ok :D  Thanks for giving it a try :)